paper/tex/methods.tex

```diff
@@ -170,74 +171,97 @@ \subsubsection{GPz}
+% The last module includes also a post-processing tool, providing some statistics on the produced point estimates and PDFs.
 
+In this work, we used a hierarchical KNN to replace nondetections with values based on their neighbors.
+The usual cross-validation step was also omitted for this study.
 
 \subsubsection{SkyNet}
 \label{sec:skynet}
 %(J. Cohen-Tanugi and Hugo Tranin)
 
-\textsc{SkyNet}\footnote{\url{http://ccpforge.cse.rl.ac.uk/gf/project/skynet/}} \citep{Graff:14} is a publicly available neural network software, based on a 2nd order conjugate gradient optimization scheme \citep[see][for further details]{Graff:14}. %It has been used efficiently for redshift PDF estimates \citep{Sanchez:14,Bonnett:15,Bonnett:16}.
-
-%In the present work, we use \textsc{SkyNet} both as a regressor (for redshift point estimates) and a classifier (for PDF estimates).
-The neural network is configured as a standard multilayer perceptron with three hidden layers and one input layer with $12$ nodes (the $6$ magnitudes and their errors).
-%For the regressor, the output layer is a single linear node, while each hidden layer has $10$ nodes with a tanh activation function.
-The classifier is laid out such that the hidden layers have 20:40:40 nodes each, all rectified linear units, and the output layer has $200$ nodes (corresponding to 200 bins for the PDF) activated with a ``softmax'' function so that they automatically sum to $1$.  While previous implementations of the code, such as \citet{Sanchez:14} and \citet{Bonnett:15} (see Appendix C.3), implement a ``sliding bin'' smoothing, no such procedure was used in this study.
+\skynet \footnote{\url{http://ccpforge.cse.rl.ac.uk/gf/project/skynet/}} \citep{Graff:14} employs a neural network based on a second order conjugate gradient optimization scheme \citep[see][for further details]{Graff:14}. %It has been used efficiently for redshift PDF estimates \citep{Sanchez:14,Bonnett:15,Bonnett:16}.
+The neural network is configured as a standard multilayer perceptron with three hidden layers and one input layer with 12 nodes corresponding to the 6 photometric magnitudes and their measurement errors.
+We use \skynet\ as a regressor for \pz\ point estimation and as a classifier for \pzpdf\ estimation.
 
-To avoid over-fitting, a $30$ per cent fraction of the training set is used as validation, and the training is stopped as soon as the error rate begins to increase in the validation set. The weights are randomly initialized based on normal sampling. The error function is a standard chi-square function for the regressor, and a cross-entropy function for the classifier. Finally, the data are all whitened before processing, with magnitudes pegged to (45,45,40,35,42,42) and their errors pegged to (20,20,10,5,15,15) for $ugrizy$ filters, respectively.
+The regressor used a standard \chisq\ error function with a single linear node as the output layer and 10 nodes with a $\tanh$ activation function for each hidden layer.
```

Comment:
remove text on regressor, as it wasn't included in the final paper, only the classifier results were used.